apps/src/applab/redux/data.js

```diff
@@ -21,13 +24,22 @@ const initialState = new DataState();
 
 export default function (state = initialState, action) {
   switch (action.type) {
+    case ADD_TABLE_NAME:
+      return state.set('tableListMap', Object.assign({}, state.tableListMap, {
+        [action.tableName]: true
+      }));
     case CHANGE_VIEW:
       // Discard table data when not viewing a table, so that we don't momentarily
       // show data for the wrong table when we return to the table view.
       if (action.view !== DataView.TABLE) {
         state = state.set('tableData', {});
       }
       return state.set('view', action.view).set('tableName', action.tableName);
+    case DELETE_TABLE_NAME: {
+      let map = Object.assign({}, state.tableListMap);
+      delete map[action.tableName];
+      return state.set('tableListMap', map);
+    }
```

Comment:
the arbitrary block is valid syntax? i thought i'd tried this before and found that it wasnt. maybe i did something wrong